### PR TITLE
feat: implement CLI with cyclopts (datasets list/info/download + run)

### LINKED ISSUE #25: feat: implement CLI with cyclopts (datasets list/info/download + run)
## Summary

Implement the CLI module (`cli.py`) using cyclopts, providing four commands that make the system user-friendly: dataset discovery (`list`, `info`), data download (`download`), and pipeline execution (`run`). This replaces the current `python -m hydro_param.pipeline` entry point.

## Design Reference

- design.md §11.9: Full CLI design with expected output formats
- design.md §11.10: User workflow (CLI is the primary interface)

### `hydro-param datasets list`
Display datasets grouped by category. Show name, description, strategy, and access status:
```
Topography:
  dem_3dep_10m     USGS 3DEP Seamless DEM, 1/3 arc-second    [stac_cog, remote]

Land Cover:
  nlcd_2021        NLCD 2021 Land Cover, 30m                  [local_tiff, download required]

Climate:
  gridmet          gridMET daily meteorological data, ~4km     [climr_cat, remote]

Soils:
  polaris_100m     POLARIS soil properties, 100m              [converted_zarr, not yet available]
```

### `hydro-param datasets info <name>`
Show full registry entry including variables, CRS, and download instructions:
```
Dataset: nlcd_2021
Description: NLCD 2021 Land Cover, 30m (Collection 1.1)
Strategy: local_tiff (download required)
CRS: EPSG:5070
Variables: land_cover (categorical)

Download:
  URL: s3://usgs-landcover/...
  Size: ~1.5 GB (COG)
  Command: aws s3 cp --no-sign-request <url> .

To use in your pipeline config:
  datasets:
    - name: nlcd_2021
      source: /path/to/your/downloaded/nlcd_2021.tif
      variables: [land_cover]
```

### `hydro-param datasets download <name> [--dest <path>]`
Execute the download for datasets with a `download` block. Default dest: current directory.

### `hydro-param run <config.yml> [--registry <registry.yml>]`
Execute the pipeline. Replaces `python -m hydro_param.pipeline`.

## NLCD Multi-Year/Product Support

NLCD provides multiple products across multiple years. The registry and CLI should support this:

**Products:** Land Cover, Impervious Surface, Tree Canopy
**Years:** 2001, 2004, 2006, 2008, 2011, 2013, 2016, 2019, 2021

**Approach:** One registry entry per dataset (e.g., `nlcd`) with all years and products listed. The `download` block becomes a list of available files keyed by year/product. The user selects which years and variables to download via the CLI:

```bash
# List available NLCD products and years
hydro-param datasets info nlcd

# Download specific years/products
hydro-param datasets download nlcd --years 2019,2021 --variables land_cover,impervious
```

The registry schema needs a `files` list (or similar) in the `download` block so the CLI can present all available options and the user can select a subset. This requires expanding the `DownloadInfo` model from #23 to support multiple downloadable files.

**Registry sketch:**
```yaml
nlcd:
  description: "NLCD Land Cover, Impervious, and Tree Canopy (30m)"
  strategy: local_tiff
  crs: "EPSG:5070"
  category: land_cover
  download:
    files:
      - year: 2021
        variable: land_cover
        url: "s3://usgs-landcover/collection-1-1/nlcd_2021_land_cover_l48_20230630.tif"
        size_gb: 1.5
      - year: 2021
        variable: impervious
        url: "s3://usgs-landcover/collection-1-1/nlcd_2021_impervious_l48_20230630.tif"
        size_gb: 1.2
      - year: 2019
        variable: land_cover
        url: "s3://usgs-landcover/..."
        size_gb: 1.5
      # ... more years/products
  variables:
    - name: land_cover
      band: 1
      long_name: "NLCD land cover class"
      categorical: true
    - name: impervious
      band: 1
      long_name: "Percent developed imperviousness"
      categorical: false
    - name: tree_canopy
      band: 1
      long_name: "Percent tree canopy cover"
      categorical: false
```

### 1. Add cyclopts dependency (`pyproject.toml`)
Add `cyclopts` to `[project.dependencies]` and the relevant pixi feature.

### 2. Expand `DownloadInfo` schema (`dataset_registry.py`)
Add support for a `files` list with year/variable/URL tuples alongside the existing single-URL `url` field (backward compat with #23).

### 3. Create `cli.py` (`src/hydro_param/cli.py`)
Single module with cyclopts app, four command handlers, and registry loading.

### 4. Register entry point (`pyproject.toml`)
```toml
[project.scripts]
hydro-param = "hydro_param.cli:main"
```

### 5. Update `__main__.py`
Wire `__main__.py` to the CLI entry point (or keep as fallback).

### 6. Expand `datasets.yml` NLCD entry
Replace the single `nlcd_2021` entry with a comprehensive `nlcd` entry listing all years and products.

### 7. Tests
- Test `datasets list` output format
- Test `datasets info` for known datasets (with and without download block)
- Test `datasets info` for unknown dataset (error handling)
- Test `datasets download` with a mock (don't actually download)
- Test `datasets download` with year/variable selection
- Test `run` invokes the pipeline (mock the pipeline stages)
- Test `--registry` flag

## Acceptance Criteria

- [ ] `cyclopts` added as a dependency
- [ ] `cli.py` module with all four commands
- [ ] Entry point registered: `hydro-param = "hydro_param.cli:main"`
- [ ] `datasets list` groups by category with access status
- [ ] `datasets info` shows full details including download instructions
- [ ] `datasets download` supports `--years` and `--variables` flags for multi-file datasets
- [ ] `run` replaces `python -m hydro_param.pipeline`
- [ ] NLCD registry entry expanded with all years/products
- [ ] Tests for all commands
- [ ] `pixi run -e dev check` passes

## Related Issues

- Depends on: #23 (download block), #24 (source override)
- Consumes: registry `download` block, pipeline config `source` override

### PR DESCRIPTION
**Pull request by rmcd-mscb**
## Summary
- Add cyclopts-based CLI with 4 commands: `datasets list`, `datasets info`, `datasets download`, `run`
- Expand `DownloadInfo` schema with `DownloadFile` model for multi-file datasets (year/variable/url)
- Replace single `nlcd_2021` registry entry with comprehensive `nlcd` entry (16 files across 9 years, 3 products)
- Register `hydro-param` console script entry point in pyproject.toml
- Rewire `__main__.py` to use CLI instead of `pipeline.main()`

## Test plan
- [x] 17 new CLI tests covering all commands and edge cases
- [x] Updated registry tests for multi-file `DownloadInfo` and renamed NLCD entry
- [x] All 103 tests pass
- [x] `pixi run -e dev check` (lint + format + mypy + tests) passes
- [x] `pixi run -e dev pre-commit` passes
- [x] Smoke tested: `hydro-param datasets list`, `hydro-param datasets info nlcd`, `hydro-param datasets info dem_3dep_10m`

Closes #25

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/hydro_param/dataset_registry.py [hunk 1]
```diff
@@ -34,13 +34,23 @@ class DerivedVariableSpec(BaseModel):
...
+
 class DownloadInfo(BaseModel):
     """Download provenance for datasets requiring local staging."""
 
-    url: str
+    url: str = ""
     size_gb: float | None = None
     format: str = ""
     notes: str = ""
+    files: list[DownloadFile] = []
```
[the remark this answers is not in the document]
Comment (rmcd-mscb) [reply src/hydro_param/dataset_registry.py]: Fixed — added a `@model_validator(mode="after")` on `DownloadInfo` that rejects instances with neither `url` nor `files`. Updated the test from `test_download_info_url_defaults_empty` to `test_download_info_requires_url_or_files` which now asserts the `ValidationError`. See f2e42d5.

File: src/hydro_param/data_access.py [hunk 2]
```diff
@@ -295,11 +295,18 @@ def fetch_local_tiff(
...
-            if entry.download.size_gb:
-                msg += f"\nExpected size: ~{entry.download.size_gb} GB"
-            if entry.download.notes:
-                msg += f"\n{entry.download.notes.strip()}"
+            if entry.download.files:
+                msg += (
+                    f"\n\nThis dataset has {len(entry.download.files)} "
+                    f"downloadable files. Run:\n"
+                    f"  hydro-param datasets info <name>"
+                )
```
[the remark this answers is not in the document]
Comment (rmcd-mscb) [reply src/hydro_param/data_access.py]: Fixed — `fetch_local_tiff` now accepts a `dataset_name` keyword argument (defaults to `"unknown"`). The pipeline call site passes `ds_req.name`, so error messages now print the actual dataset name and actionable CLI command. See f2e42d5.
